New inspector implementation

vpereverzev opened the pull request:
(General - Notation - Playback)

Ссылка на макеты - https://www.figma.com/file/l0TQUrNFZmEfrZnfHhFDfb/Musescore-Notation-App-MASTER?node-id=0%3A1

vpereverzev committed:
Added 'inspectors' module

vpereverzev committed:
Added separate resource file .qrc for 'inspectors' module

vpereverzev committed:
Added 'inspectors' module setup

vpereverzev committed:
Moved all related to inspectors stuff  to module 'inspectors'

vpereverzev committed:
Got rid of direct mscore's usage from inspectors module

vpereverzev committed:
Moved miconengine to global module

vpereverzev committed:
Created a separate resources .qrc file for inspectors module with it's own icons

vpereverzev committed:
Now inspectors module manages it's icons by itself

vpereverzev committed:
Enabled QML DEBUG

vpereverzev committed:
Temporarily set up a GUI module to separate icon engine and to research the possibility to setup QmlPlugin with general components

vpereverzev committed:
Added to elements the way to get their representative "base" element, like a chord for the note

vpereverzev committed:
included precompiled headers to the inspectors module

vpereverzev committed:
SImplified a bit the undochangeproperty algorithm and fixed issue with 3-ple inspector updating

vpereverzev committed:
Attempt to untie the inspector.cpp from mscore using signals + slots

vpereverzev committed:
Explicitly set property style flags for chord property change

vpereverzev committed:
Implemented InspectorFormWidget as a container for the new QML-based inspector implementation

vpereverzev committed:
Temporarily disabled the old implementation of the inspector and replaced it with a new one

vpereverzev committed:
Implemented the main inspector qml form

vpereverzev committed:
Implemented CheckBox qml component

vpereverzev committed:
Implemented ExpandableBlank qml component

vpereverzev committed:
Implemented FlatButton qml component

vpereverzev committed:
Implemented IncrementalPropertyControl

vpereverzev committed:
Implemented TabPanel qml component

vpereverzev committed:
Implemented FlatRadioButton general qml component

vpereverzev committed:
Implemented FlatToogleButton general qml component

vpereverzev committed:
Implemented FocusableItem general qml component

vpereverzev committed:
Implemented RadioButtonGroup general qml component

vpereverzev committed:
Implemented StyledComboBox general qml component

vpereverzev committed:
Implemented StyledIcon general qml component

vpereverzev committed:
Implemented StyledPopup general qml component

vpereverzev committed:
Implemented StyledTextLabel general qml component

vpereverzev committed:
Implemented inspector's menu

vpereverzev committed:
Implementation of the DoubleInputValidator (author @H.Chang) for the IncrementalPropertyControl

vpereverzev committed:
Added the bunch of icons for the new inspector

vpereverzev committed:
Implemented the PropertyItem as a basic entity for the inspectors properties

vpereverzev committed:
Implemented the AbstractInspectorModel as a base class for the entire set of inspector's models

vpereverzev committed:
Implemented interface for the elementrepositoryservice

vpereverzev committed:
Implemented ElementRepositoryService as a main elements provider for inspectors models

vpereverzev committed:
Implemented inspectorListModel, the listmodel that will handle main section models in the inspector (General and Notation for now)

vpereverzev committed:
Implemented the GeneralSettingsModel as a handler for general element's properties (like visible, small, etc.)

vpereverzev committed:
Implemented the GeneralInspectorView

vpereverzev committed:
Implemented NotationInspectorProxyModel as a proxy-representative for the notation inspector models

vpereverzev committed:
Implemented NoteHeadSettingsModel for the Notehead inspector's section

vpereverzev committed:
Implemented the Notation inspector popup view with the content

vpereverzev committed:
Implemented the NoteHead inspector view

vpereverzev committed:
Added the separate method for the updating note head group

vpereverzev committed:
Removed unused method from Note class

vpereverzev committed:
Added for the chords possibility to process "PLAY" property of children elements (notes, tremolo, arpeggio)

vpereverzev committed:
Implemented the model for Beam inspector section

vpereverzev committed:
Extracted the method for processing feathering of the beams

vpereverzev committed:
Implemented the Beam inspector view section

vpereverzev committed:
Implemented the Stem and Hook models for the "STEM" inspector section

vpereverzev committed:
Implemented the Stem inspector's section view

vpereverzev committed:
Implemented the Playback proxy model as a proxy-representative for playback inspectors models

vpereverzev committed:
Implemented the Playback inspector's section popup view with the relative content

vpereverzev committed:
Implemented the Arpeggio playback model

vpereverzev committed:
Implemented the Arpeggio playback section view

vpereverzev committed:
Implemented the Note playback inspector model

vpereverzev committed:
Implemented the Note playback section view

vpereverzev committed:
Implemented the Breaths playback model

vpereverzev committed:
Implemented the Breaths & pauses playback section view

vpereverzev committed:
Implemented the Hairpin playback model

vpereverzev committed:
Implemented the Hairpin playback view section

vpereverzev committed:
Implemented the Glissando playback model

vpereverzev committed:
Implemented the Dynamics playback view section

vpereverzev committed:
Implemented the Dynamics playback model

vpereverzev committed:
Limited the possibility to change velocity change property for "pf, sfp, fp and sfpp" dynamics

vpereverzev committed:
Implemented Glissando playback model

vpereverzev committed:
Implemented the Fermata playback model

vpereverzev committed:
Implemented the Fermata playback section view

vpereverzev committed:
Registered some C++ enums from the inspector to QML

vpereverzev committed:
Set the default font size to 12

vpereverzev committed:
Clean up includes in the inspector module

igorkorsukov commented on gui/guisetup.cpp:
```diff
@@ -0,0 +1,46 @@
+#include "guisetup.h"
+
+
+GuiSetup::GuiSetup()
```
А что этот модуль доожен содеражить? 
И что должно быть в `global/gui`?

vpereverzev replied:
я пока удалил за ненадобностью, создавал его для небольшого эксперимента

В нем предполагалось содержать QML Engine, global style и general qml компоненты
Сейчас не до конца уверен с QML engine, где именно он должен находиться и сетапиться. Но место под общие компоненты все же нужно будет придумать позже.

igorkorsukov commented on inspectors/inspectorssetup.cpp:
```diff
@@ -0,0 +1,59 @@
+    qmlRegisterUncreatableType<HairpinTypes>("MuseScore.Inspectors", 3, 3, "Hairpin", "Not creatable as it is an enum type");
+    qmlRegisterUncreatableType<ArticulationTypes>("MuseScore.Inspectors", 3, 3, "Articulation", "Not creatable as it is an enum type");
+    qmlRegisterUncreatableType<DynamicTypes>("MuseScore.Inspectors", 3, 3, "Dynamic", "Not creatable as it is an enum type");
+    qmlRegisterUncreatableType<GlissandoTypes>("MuseScore.Inspectors", 3, 3, "Glissando", "Not creatable as it is an enum type");
```
может быть не удобно, что названия с++ классов и qml типов разные, будет сложнее искать.  
и возможно может быть путаница (в голове) между объектами дом модели и эти типами qml (они называются как будто они элементы дом модели)

vpereverzev replied:
соглашусь, добавлю "Types" в qml представление

igorkorsukov commented on inspectors/models/abstractinspectormodel.cpp:
```diff
@@ -0,0 +1,309 @@
+
+    emit elementsModified();
+
+    m_elementList.at(0)->triggerLayout();
```
не может ли быть тут пусто? 
почему вызывается `triggerLayout` только у первого элемента? (нужен коммент)
в `onPropertyValueChanged` вызывается у всех, поэтому возникают вопросы

vpereverzev replied:
от пустого списка защитит if (isEmpty())

но на самом деле, я просто не заметил этот вызов, когда изменял onPropertyValueChanged

triggerLayout нужно поместить в цикл

igorkorsukov commented on inspectors/models/abstractinspectormodel.h:
```diff
@@ -0,0 +1,85 @@
+    QVariant valueToElementUnits(const Ms::Pid& pid, const QVariant& value, const Ms::Element* element) const;
+    QVariant valueFromElementUnits(const Ms::Pid& pid, const QVariant& value, const Ms::Element* element) const;
+
+    IElementRepositoryService* m_repository;
```
лучше всегда инициализировать `IElementRepositoryService* m_repository{nullptr};

vpereverzev replied:
да, пропустил по невнимательности, спасибо

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.cpp:
```diff
@@ -0,0 +1,92 @@
+#include "generalsettingsmodel.h"
+
+GeneralSettingsModel::GeneralSettingsModel(QObject* parent, IElementRepositoryService* repository) : AbstractInspectorModel(parent, repository)
```
лучше `:` и то что дальше на следующей строке писать, а то в github вылазит за границы :) 
(можно в QtCreator включить отображения границы, установить границу на 120)

vpereverzev committed:
Removed empty experimental module

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.cpp:
```diff
@@ -0,0 +1,92 @@
+
+void GeneralSettingsModel::requestElements()
+{
+    m_elementList = m_repository->takeAllElements();
```
а метод `takeAllElements` отдаёт элементы и удаляет их из репозитория? Подобно методу `QMap::take` ?

vpereverzev replied:
нет, временем жизни не управляет, думаешь возникнет диссонанс?
можно take на get заменить или вообще убрать глагол

igorkorsukov replied:
думаю можно убрать глагол.

vpereverzev committed:
Fix compilation errors after the rebase onto master

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.cpp:
```diff
@@ -0,0 +1,92 @@
+    m_isPlayable = buildPropertyItem(Ms::Pid::PLAY);
+    m_isPlayModeAvailable = buildPropertyItem(Ms::Pid::PLAY);
+    m_isSmall = buildPropertyItem(Ms::Pid::SMALL);
+    m_isSmallModeAvailable = buildPropertyItem(Ms::Pid::SMALL);
```
названия у переменных как будто они булевые, вводят в заблуждение

vpereverzev replied:
а эти свойства на самом деле булевы (QVariant(bool)), названием как раз это и хотел передать

igorkorsukov replied:
думаю лучше всётаки добавить вконце `Item`
`m_isPlayableItem`
`m_isPlayModeAvailableItem`
`m_isSmallItem`
и тд

igorkorsukov replied:
А может и не надо... смотрю дальше ПР

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.cpp:
```diff
@@ -0,0 +1,92 @@
+void GeneralSettingsModel::resetProperties()
+{
+    m_isVisible->resetToDefault();
+    m_isAutoPlaceAllowed->resetToDefault();
```
а почему тут не все, а только эти? может коммент нужен?

vpereverzev replied:
поторопился и упустил ресет остальных
хорошо, что ты заметил

igorkorsukov replied:
Может завести член класса `_allItems`, туда поместить все item и использовать её там, где нужно все (чтобы не завбвать добавлять в нужные места и явно видеть что тут нужно сделать действие над всеми)

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.cpp:
```diff
@@ -0,0 +1,92 @@
+    return m_isPlayModeAvailable;
+}
+
+PropertyItem* GeneralSettingsModel::isSmall() const
```
названия методов как будто они возвращают булевые значения, вводит в заблуждение

igorkorsukov commented on inspectors/models/general/generalsettingsmodel.h:
```diff
@@ -0,0 +1,55 @@
+    PropertyItem* m_isSmall = nullptr;
+    PropertyItem* m_isSmallModeAvailable = nullptr;
+
+    PlaybackProxyModel* m_playbackProxyModel = nullptr;
```
надо нам со стилем определиться. 
я предпочитаю `PlaybackProxyModel* m_playbackProxyModel {nullptr};` и вроде так в большинствей частей проекта. 
Тебе нравиться через `=`?

vpereverzev replied:
да, обсудим
мне больше полюбилось =, чуть меньше механических действий, когда пишешь

vpereverzev committed:
Changed the feathering beam increment step to 0.1 and added limitations for incrementalpropertycontrol

vpereverzev committed:
Commit changes in IncrementalInputControl by pressing Enter/Return

vpereverzev committed:
Renamed properties for qt 5.14 compatibility

vpereverzev committed:
Fixed nullptr connect between the models

Harmoniker1 committed:
Add top/bottom/decimal properties to DoubleInputValidator

+ create `IntInputValidator`

vpereverzev committed:
Renamed StyledComboBox property handlers for Qt 5.14 compatibility

igorkorsukov commented on global/gui/miconengine.cpp:
```diff
@@ -13,12 +13,16 @@
 
+QString MIconEngine::iconDirPath = QString();
+
 class MIconEnginePrivate : public QSharedData
```
Не место ему в `global` (этот комент как закладка чтобы обсудить этот вопрос)

igorkorsukov commented on inspectors/inspectoriconloader.cpp:
```diff
@@ -0,0 +1,21 @@
+
+QMap<QString, QIcon*> InspectorIconLoader::m_iconCache = QMap<QString, QIcon*>();
+
+QIcon *InspectorIconLoader::icon(const char *iconName)
```
вероятно этот метод должен быть `const`

igorkorsukov commented on inspectors/inspectoriconloader.h:
```diff
@@ -0,0 +1,30 @@
+#define INSPECTORICONLOADER_H
+
+
+class InspectorIconLoader
```
Надо обсудить политику использования неймспесов

igorkorsukov commented on inspectors/interfaces/ielementrepositoryservice.h:
```diff
@@ -0,0 +1,23 @@
+public:
+    virtual ~IElementRepositoryService() = default;
+
+    virtual QObject* getQObject() = 0;
```
Думаю надо будет сдёрнуть с МТ каналы :)

igorkorsukov commented on inspectors/models/abstractinspectormodel.h:
```diff
@@ -0,0 +1,85 @@
+    void updateProperties();
+
+    QString m_title;
+    InspectorModelType m_type;
```
лучше всегда инициализировать `InspectorModelType m_type { UNDEFINED }`

igorkorsukov commented on inspectors/models/general/playback/internal_models/arpeggioplaybackmodel.cpp:
```diff
@@ -0,0 +1,33 @@
+#include "arpeggioplaybackmodel.h"
+
+ArpeggioPlaybackModel::ArpeggioPlaybackModel(QObject* parent, IElementRepositoryService* repository) : AbstractInspectorModel(parent, repository)
```
Перенести `:` и то что за ней на др строчку

igorkorsukov commented on inspectors/models/general/playback/internal_models/arpeggioplaybackmodel.h:
```diff
@@ -0,0 +1,26 @@
+
+    Q_PROPERTY(PropertyItem* stretch READ stretch CONSTANT)
+public:
+    explicit ArpeggioPlaybackModel(QObject* parent, IElementRepositoryService* repository);
```
Я вот думаю, может термины "репозоторий", "сервис" не из мира этого приложения, смотрятся чужеродно :)   
Может проще, например вместо `IElementRepositoryService` назвать `IDomElementAccessor`?

Harmoniker1 replied:
"Dom" and "Accessor" look alien too to me :-)

igorkorsukov replied:
Dom - Domain Object Model (this is libmscore :) )
Maybe not add `Dom`, but just `IElementsAccessor`

vpereverzev replied:
Я думаю, что сейчас любое название будет +- чужеродным, потому что сейчас ничего такого не используется. Можно по-тихоньку вводить терминологию паттернов, они по-идее должны быть общеизвестными, либо можно быстро нагуглить, что делает паттерн с таким термином

igorkorsukov replied:
"репозиторий" - это обычно о хранилище, т.е в данном приложении репозиторий - это чтение и запись скоров, вот что меня смущает.

igorkorsukov commented on inspectors/models/general/playback/internal_models/breathplaybackmodel.cpp:
```diff
@@ -0,0 +1,33 @@
+#include "breathplaybackmodel.h"
+
+BreathPlaybackModel::BreathPlaybackModel(QObject* parent, IElementRepositoryService* repository) : AbstractInspectorModel(parent, repository)
```
Перенести `:` на др строку

igorkorsukov commented on inspectors/models/general/playback/internal_models/dynamicplaybackmodel.cpp:
```diff
@@ -0,0 +1,83 @@
+
+#include "dynamic.h"
+
+DynamicPlaybackModel::DynamicPlaybackModel(QObject* parent, IElementRepositoryService* repository) : AbstractInspectorModel(parent, repository)
```
Перенести `:` на др строку

igorkorsukov commented on inspectors/models/general/playback/internal_models/dynamicplaybackmodel.cpp:
```diff
@@ -0,0 +1,83 @@
+    m_velocityChange = buildPropertyItem(Ms::Pid::VELO_CHANGE);
+
+    //@note readonly property, there is no need to modify it
+    m_isVelocityChangeAvailable = buildPropertyItem(Ms::Pid::DYNAMIC_TYPE, [](const int, const QVariant&){});
```
Может лучше добавить отдельный метод?, например `buildReadonlyPropertyItem`

vpereverzev replied:
подумываю завести свойство isReadOnly в PropertyItem
им бы и здесь можно было воспользоваться, и при ресете по списку PropertyItem-ов

vpereverzev committed:
Alter the "Force horizontal" behaviour to the new rule

Force horizontal now is a button and it'll align the beam position based on the further beam's point relative to the stave

vpereverzev committed:
Altered locked beam height processing, fixed a recursive handlers

igorkorsukov commented on inspectors/models/notation/beams/beamsettingsmodel.h:
```diff
@@ -0,0 +1,76 @@
+
+    Q_PROPERTY(PropertyItem* featheringHeightLeft READ featheringHeightLeft CONSTANT)
+    Q_PROPERTY(PropertyItem* featheringHeightRight READ featheringHeightRight CONSTANT)
+    Q_PROPERTY(int featheringMode READ featheringMode WRITE setFeatheringMode NOTIFY featheringModeChanged)
```
А может быть всё же лучше добавить `Item` в конец, а то вот в таких случая не понятно будет в `qml` какой тип у свойства

igorkorsukov commented on inspectors/models/notation/beams/internal/beammodelistmodel.h:
```diff
@@ -0,0 +1,50 @@
+
+private:
+    struct BeamTypesData {
+        int mode;
```
лучше всегда инициализировать, например 0

igorkorsukov commented on inspectors/models/notation/beams/internal/beammodelistmodel.h:
```diff
@@ -0,0 +1,50 @@
+    QHash<int, QByteArray> m_roleNames;
+
+    QVector<BeamTypesData> m_beamTypesDataList;
+    int m_selectedTypeIndex;
```
лучше всегда инициализировать, например 0 или -1

igorkorsukov commented on inspectors/models/notation/beams/internal/beammodesmodel.cpp:
```diff
@@ -0,0 +1,79 @@
+
+void BeamModesModel::resetProperties()
+{
+    m_mode->resetToDefault();
```
не ресетится свойство `m_isFeatheringAvailable`
Если это ошибка (забыл), то лучше в самом деле ввести `m_allItems` на уровне базовго класса и через него делать массовые операции. А также тогда метод `resetProperties` имел бы базовую реализацию

igorkorsukov commented on inspectors/models/notation/noteheads/noteheadsettingsmodel.cpp:
```diff
@@ -0,0 +1,126 @@
+{
+    createProperties();
+
+    setTitle(tr("Head"));
```
надо бы подумать, чтобы явно указывать контекст перевода, а то в `.ts` файле будет куча этих моделей... 
а было бы хорошо, что бы контекст был один - `inspector`  
Можно сделать как в МТ, достаточно удобно

igorkorsukov commented on inspectors/models/notation/noteheads/noteheadtypesmodel.h:
```diff
@@ -0,0 +1,53 @@
+
+private:
+    struct HeadTypeData {
+        Ms::NoteHead::Group group;
```
лучше всегда инициализировать

igorkorsukov commented on inspectors/services/elementrepositoryservice.cpp:
```diff
@@ -0,0 +1,215 @@
+    for (const Ms::Element* element : m_elementList) {
+
+        if (element->type() == Ms::ElementType::CHORD) {
+            resultList << Ms::toChord(const_cast<Ms::Element*>(element));
```
зачем из `Element*` кастовать к `Chord, чтобы потом снова скастовать к `Element*`? Для проверки на тип?

vpereverzev replied:
нужды в этом нет, наверное копипастой из соседних методов залетело
хорошо заметил

igorkorsukov commented on inspectors/view/qml/InspectorForm.qml:
```diff
@@ -0,0 +1,103 @@
+
+import "common"
+import "general_inspector"
+import "notation_inspector"
```
Может лучше CamelCase? Чужеродно смотрится, особенно в контексте, что `qml` файлы названы в CamelCase

vpereverzev replied:
я скорее наоборот хотел, чтобы визуально не спутать названия папок и названия компонентов

Но в целом, почему бы и нет. Возможно даже стоит убрать надобность в черточке и удалить слово "inspector"

igorkorsukov commented on inspectors/view/qml/InspectorForm.qml:
```diff
@@ -0,0 +1,103 @@
+        height: parent.height
+        width: parent.width
+
+        model: presenterModel
```
А в чём приемущество использовать DelegateModel? а нет тут задать модель и делегата?

vpereverzev replied:
да по-сути сахар, хотелось отделить сопоставление и подбор делегатов

igorkorsukov commented on inspectors/view/qml/common/CheckBox.qml:
```diff
@@ -0,0 +1,93 @@
+
+    property alias text: label.text
+
+    signal clicked
```
Думаю лучше `signal clicked()` - воспринимается лучше

vpereverzev replied:
автоформатирование креатора настойчиво убирает скобки если нет аргументов

igorkorsukov replied:
хмм, ок

igorkorsukov commented on inspectors/view/qml/common/CheckBox.qml:
```diff
@@ -0,0 +1,93 @@
+                sourceSize.width: 12
+                sourceSize.height: 12
+
+                source: root.isIndeterminate ? "qrc:/resources/icons/indeterminate_check_mark.svg" : "qrc:/resources/icons/check-mark.svg"
```
Думаю нужно сделать глобальную функцию которая будет возвращать путь по названию иконки, чтото типа `ui.icon(...)`
Или свой ImageProvider

vpereverzev replied:
да, чуть позже собирался что-то подобное реализовывать, тоже посещали такие мысли

igorkorsukov commented on inspectors/view/qml/general_inspector/playback/internal/DynamicsExpandableBlank.qml:
```diff
@@ -0,0 +1,157 @@
+                model: [
+                    { text: "Staff", value: Dynamic.SCOPE_STAFF },
+                    { text: "Single instrument", value: Dynamic.SCOPE_SINGLE_INSTRUMENT },
+                    { text: "All instruments", value: Dynamic.SCOPE_ALL_INSTRUMENTS }
```
вероятно переводимые должны быть?

igorkorsukov commented on inspectors/view/qml/general_inspector/playback/internal/HairpinsExpandableBlank.qml:
```diff
@@ -0,0 +1,80 @@
+            spacing: 8
+
+            StyledTextLabel {
+                text: qsTr("Velocity change")
```
Также как в cpp, нужно решить как декларировать переводы, с таким способом будет qml файл в качестве контекста, и в лингвусе соотвественно будет много .qml файлов. И при переименовании файла или переносе строки в др место перевод будет не ломаться, нужно будет заново переводить. 
Было бы хорошо чтобы все эти переводимые строки были в одном контексте, 
для этого есть варианты: 
1. Вынести весь перевод в одно место (js или qml) файл - не очень удобно
2. Добавить свою функцию перевода, будет выглядить так `qsTrc("inspector", "Velocity change")`

igorkorsukov commented on inspectors/view/qml/general_inspector/playback/internal/PausesExpandableBlank.qml:
```diff
@@ -0,0 +1,39 @@
+        IncrementalPropertyControl {
+            iconMode: iconModeEnum.hidden
+
+            measureUnitsSymbol: "s"
```
вероятно должен быть переводимый

igorkorsukov commented on inspectors/view/ui/alignSelect.cpp:
```diff
@@ -36,13 +36,13 @@ AlignSelect::AlignSelect(QWidget* parent)
+      alignVCenter->setIcon(*InspectorIconLoader::icon(InspectorIconLoader::TEXT_ALIGN_V_CENTER_ICON));
+      alignTop->setIcon(*InspectorIconLoader::icon(InspectorIconLoader::TEXT_ALIGN_V_TOP_ICON));
+      alignBaseline->setIcon(*InspectorIconLoader::icon(InspectorIconLoader::TEXT_ALIGN_V_BASELINE_ICON));
+      alignBottom->setIcon(*InspectorIconLoader::icon(InspectorIconLoader::TEXT_ALIGN_V_BOTTOM_ICON));
```
Думаю лучше константы названий иконок прописать в отдельном классе, например чтобы было так: 
`InspectorIconLoader::icon(InspectorIcon::TEXT_ALIGN_V_BOTTOM_ICON)`

vpereverzev replied:
По-сути это все не нужно, если старую реализацию не поддерживать

igorkorsukov commented on inspectors/view/ui/fretcanvas.h:
```diff
@@ -34,11 +35,11 @@ enum class FretDotType : signed char;
-      int cfret            { 0 };
+      FretDiagram* diagram;
+      int cstring;
+      int cfret;
```
теперь не инициализируются - это следствие ребайза?

vpereverzev replied:
ага, вручную файл не трогал
над вернуть

igorkorsukov commented on libmscore/chord.cpp:
```diff
@@ -2725,6 +2762,7 @@ QVariant Chord::getProperty(Pid propertyId) const
             case Pid::NO_STEM:        return noStem();
             case Pid::SMALL:          return small();
             case Pid::STEM_DIRECTION: return QVariant::fromValue<Direction>(stemDirection());
+            case Pid::PLAY: return isChordPlayable();
```
выравнивание

igorkorsukov commented on libmscore/element.h:
```diff
@@ -197,6 +197,7 @@ class Element : public ScoreElement {
       const Measure* findMeasure() const;
       MeasureBase* findMeasureBase();
       const MeasureBase* findMeasureBase() const;
+      virtual Element* elementBase() const { return const_cast<Element*>(this); }
```
нужен коммент что такое `elementBase`

igorkorsukov commented on libmscore/hook.h:
```diff
@@ -28,10 +28,10 @@ class Hook final : public Symbol {
+      virtual Hook* clone() const override        { return new Hook(*this); }
+      Element* elementBase() const override;
+      virtual qreal mag() const override          { return parent()->mag(); }
+      virtual ElementType type() const override   { return ElementType::HOOK; }
```
Давай не будем ставить `virtual` у `override` методов (у меня было несколько массовых ПР где я это убирал)

igorkorsukov commented on libmscore/note.h:
```diff
@@ -464,12 +466,11 @@ class Note final : public Element {
+      virtual QVariant getProperty(Pid propertyId) const override;
+      virtual bool setProperty(Pid propertyId, const QVariant&) override;
+      virtual QVariant propertyDefault(Pid) const override;
+      virtual QString propertyUserValue(Pid) const override;
```
Давай не будем ставить virtual у override методов (у меня было несколько массовых ПР где я это убирал)

vpereverzev replied:
после ребейза наверное попали, я такое тоже не одобряю

igorkorsukov commented on libmscore/stem.h:
```diff
@@ -34,12 +34,13 @@ class Stem final : public Element {
+      virtual void draw(QPainter*) const override;
+      virtual bool isEditable() const override    { return true; }
+      virtual void layout() override;
+      virtual void spatiumChanged(qreal /*oldValue*/, qreal /*newValue*/) override;
```
Давай не будем ставить virtual у override методов (у меня было несколько массовых ПР где я это убирал)

igorkorsukov commented:
Думаю нужно разделить перенос старого инспектора и создание нового, сделать два разных ПР и ПР с переносом влить. Или не трогать старый инспектор, и в этом ПР вернуть его на место. 
Это нужно для двух целей: 
* Надо чтобы старый и новый инспектор были физически друг с другом не связано (даже лучше продублировать какойто код, если требуется)
* Этот ПР очень большой, его и без переноса старого сложно ревьювить :)

vpereverzev committed:
Got rid of unnecessary cast

vpereverzev committed:
Added the SeparatorLine general qml component

vpereverzev committed:
Returned the hiding/showing behaviour for "feathered beams" stuff

vpereverzev committed:
Reload beam's model properties after the change of beam type or "force horizontal" apply

vpereverzev committed:
Added members variables initialization, removed virtual keyword for overriden methods and made other style fixes by review comments

vpereverzev committed:
Automatically select all text within IncrementalPropertyControl when it recieves focus

vpereverzev committed:
Added forgotten beam type value

vpereverzev committed:
Protected from the old crash when updating the grips

vpereverzev committed:
Now selected elements that are not supporting the selected property don't influence on it's indeterminate state

vpereverzev committed:
Expanded the list of dynamic's types that can process the velocity change property

vpereverzev committed:
Repair missing files after the rebase

vpereverzev committed:
Update collect_artifacts.yml

vpereverzev committed:
Update collect_artifacts.yml

vpereverzev committed:
Update collect_artifacts.yml
